tests/src/Integration/Engine/AnnotationProcessingTest.php

```diff
@@ -0,0 +1,136 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\Tests\rules\Integration\Engine\AnnotationProcessingTest.php.
+ */
+
+namespace Drupal\Tests\rules\Integration\Engine;
+
+use Drupal\Tests\rules\Integration\RulesIntegrationTestBase;
+use Drupal\Core\Session\SessionManagerInterface;
+
+/**
+ * Test auto saving of variables after Rules execution.
+ *
+ * @group rules
+ */
+class AnnotationProcessingTest extends RulesIntegrationTestBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function setUp() {
+    parent::setUp();
+    $this->enableModule('user');
+    $this->sessionManager = $this->prophesize(SessionManagerInterface::class);
+    $this->container->set('session_manager', $this->sessionManager->reveal());
+  }
+
+  /**
+   * Tests if our ContextDefiniton annotations are correctly processed.
+   *
+   * @dataProvider provideRulesPlugins
+   *
+   * @param array $params
+   *   An associative array containing type information from Rules plugins.
+   *
+   * @group rules
+   */
+  public function testCheckConfiguration($params) {
+    $plugin = NULL;
+
+    switch ($params['type']) {
+      case 'action':
+        $plugin = $this->actionManager->createInstance($params['plugin']);
+        break;
+
+      case 'condition':
+        $plugin = $this->conditionManager->createInstance($params['plugin']);
+        break;
+    }
+
+    $this->assertNotNull($plugin, "{$params['type']} plugin {$params['plugin']} loads");
+    if (empty($plugin)) {
+      return;
+    }
```

Comment:
why is that if() necessary? I think phpunit already stops when an assertion fails.